src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Azure/Functions/AzureFunctionCommon.cs

```diff
@@ -0,0 +1,112 @@
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Azure.Functions
+{
+    internal static class AzureFunctionCommon
+    {
+        public const string OperationName = "azure.function";
+        public static readonly IntegrationInfo IntegrationId = IntegrationRegistry.GetIntegrationInfo(nameof(IntegrationIds.AzureFunction));
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(AzureFunctionCommon));
+
+        public static CallTargetState OnMethodBegin<TTarget>(TTarget instance, IFunctionInstance instanceParam)
+        {
+            var tracer = Tracer.Instance;
+
+            if (tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                var scope = CreateScope(tracer, instanceParam);
+                return new CallTargetState(scope);
+            }
+
+            return CallTargetState.GetDefault();
+        }
+
+        public static TResult OnMethodEnd<TTarget, TResult>(TTarget instance, TResult result, Exception exception, CallTargetState state)
+        {
+            Scope scope = state.Scope;
+
+            if (scope is null)
+            {
+                return result;
+            }
+
+            try
+            {
+                if (exception != null)
+                {
+                    scope.Span.SetException(exception);
+                }
+            }
+            finally
+            {
+                scope.Dispose();
+            }
+
+            return result;
+        }
+
+        internal static Scope CreateScope(Tracer tracer, IFunctionInstance instanceParam)
+        {
+            Scope scope = null;
+
+            try
+            {
+                var triggerType = "Unknown";
+                switch (instanceParam.Reason)
+                {
+                    case AzureFunctionExecutionReason.HostCall:
+                        // Defer the span management to the AspNetCoreDiagnosticObserver
+                        return null;
+                    case AzureFunctionExecutionReason.AutomaticTrigger:
+                        triggerType = "Timer";
+                        break;
+                    case AzureFunctionExecutionReason.Dashboard:
+                        triggerType = "Dashboard";
+                        break;
+                }
+
+                var functionName = instanceParam.FunctionDescriptor.ShortName;
+                var fullName = instanceParam.FunctionDescriptor.FullName;
+
+                var resourceName = $"{triggerType} {functionName}";
+
+                var tags = new AzureFunctionTags
+                {
+                    TriggerType = triggerType,
+                    ShortName = functionName,
+                    FullName = fullName,
+                    InstrumentationName = IntegrationId.Name
+                };
+
+                tags.SetAnalyticsSampleRate(IntegrationId, tracer.Settings, enabledWithGlobalSetting: false);
```

Comment:
All web server integrations should use `enabledWithGlobalSetting: true`. I wasn't sure if we should consider this a web integration, but then I saw you set `scope.Span.Type = SpanTypes.Web` below.

(level of care low since App Analytics has been deprecated for a while and should not be used with new setups)

Reply:
This has been changed to "Serverless"